helix-core/src/main/java/org/apache/helix/common/caches/InstanceMessagesCache.java

```diff
@@ -148,9 +152,20 @@ public boolean refresh(HelixDataAccessor accessor, Map<String, LiveInstance> liv
     LOG.info(
         "END: InstanceMessagesCache.refresh(), {} of Messages read from ZooKeeper. took {} ms. ",
         newMessageKeys.size(), (System.currentTimeMillis() - startTime));
+
+    refreshStaleMessageCache();
     return true;
   }
 
+  public Map<String, Map<String, Message>> getStaleMessageCache() {
```

Comment:
I think we have this method mainly for supporting the test. Otherwise, we can have the getStaleMessageForInstance() here, right?

My suggestion is that we should not change our business logic in any way because of the test requirement. So I will say, let's move the getStaleMessageForInstance() implementation here. And leave this method as @VisibleForTesting only.

Reply:
Or, if you can make the test to check InstanceMessagesCache only, that would be even better. But that is optional.

Reply:
Tried to only use `getStaleMessageCache()` in this class. But it's difficult to change the test to check this class directly.  So keep the function in two classes.